src/reactionsystem_conversions.jl

```diff
@@ -89,41 +111,52 @@ function assemble_oderhs(rs, ispcs; combinatoric_ratelaws = true, remove_conserv
     rhsvec
 end
 
-function assemble_drift(rs, ispcs; combinatoric_ratelaws = true, as_odes = true,
-        include_zero_odes = true, remove_conserved = false)
-    rhsvec = assemble_oderhs(rs, ispcs; combinatoric_ratelaws, remove_conserved)
+# For a `ReactionSystem`, creates a vector with all the differential equations corresponding to
+# the ODEs generated by the reaction rate equation. The `as_odes` argument determine whether the LHS should
+# be differential (when generating e.g. a `ODESystem`) or `0` (when generating e.g. a NonlinearSystem).
+function assemble_drift(rs, ind_sps; combinatoric_ratelaws = true, as_odes = true,
+                        include_zero_odes = true, remove_conserved = false)
+    rhsvec = assemble_oderhs(rs, ind_sps; combinatoric_ratelaws, remove_conserved)
     if as_odes
         D = Differential(get_iv(rs))
         eqs = [Equation(D(x), rhs)
-               for (x, rhs) in zip(ispcs, rhsvec) if (include_zero_odes || (!_iszero(rhs)))]
+               for (x, rhs) in zip(ind_sps, rhsvec) if (include_zero_odes || (!_iszero(rhs)))]
     else
         eqs = [Equation(0, rhs) for rhs in rhsvec if (include_zero_odes || (!_iszero(rhs)))]
     end
     eqs
 end
 
-# this doesn't work with constraint equations currently
-function assemble_diffusion(rs, sts, ispcs; combinatoric_ratelaws = true,
-        remove_conserved = false)
+# this doesn't work with coupled equations currently
+# assemblies the diffusion equations terms used in the chemical Langevin equation SDE
+function assemble_diffusion(rs, us, ind_sps; combinatoric_ratelaws = true,
+                            remove_conserved = false)
     # as BC species should ultimately get an equation, we include them in the noise matrix
-    num_bcsts = count(isbc, get_unknowns(rs))
+    num_bcus = count(isbc, get_unknowns(rs))
```

Comment:
generally I changes `sts` and similar to `us` (to follow the convention change states to unknowns)